src/k2/dto/SKVRecord.h

```diff
@@ -90,7 +90,7 @@ class SKVRecord {
             }
         }
 
-        throw new std::runtime_error("Schema not followed in record deserialization");
+        return std::nullopt;
```

Comment:
Do we still need this change? If we keep it, what about the case where the user tries to deserialize with the wrong type, should that also return a nullopt?

Reply:
Seems the change is reflected in the test case code as well. But not sure why we need this behavior change as well. Could you explain either online or offline?

Reply:
This is needed to support our advertised "all fields are optional" interface. Imagine these scenarios in which we need to decide if we should throw or not.
0. Key is in schema, field is set but it is of incorrect type
1. Key is in schema but field is not set
2. Key is not in schema

I propose for 0 we throw since we have this key, but the type doesn't match.
For 1. and 2. it makes more sense to me to return std::nullopt to signal "this key is not set". whether it exists in the schema or not, it doesn't change the fact that it "isn't set"

Coincidentally, :D this also makes my code a bit easier since I don't have to handle case 2 specially. With a scan filter, I have to be able to evaluate a key condition (e.g. "age" > 20) on records which
1. have the key and is of compatible type
2. have the key but it isn't of compatible type
3. have the key in schema but it isn't set
4. don't have the key in the schema

In a scan with filter, i don't think there should be a difference in how we handle 3 vs 4